cloud-provider.sh

```diff
@@ -23,13 +28,26 @@ set_azure_config() {
   # login to Azure
   az login --service-principal -u ${azure_client_id} -p ${azure_client_secret} --tenant ${azure_tenant_id} 2>&1 > /dev/null
 
-  local az_vm_nic=$(az vm nic list -g ${az_resources_group} --vm-name ${az_vm_name} | jq -r .[0].id | cut -d "/" -f 9)
-  local az_subnet_name=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 11)
-  local az_vnet_name=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 9)
-  local az_vnet_resource_group=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 5)
+  if [ -z "$az_resources_group" ] ; then
+    az_resources_group="$az_vm_resources_group"
+  fi
+
+  local az_vm_nic=$(az vm nic list -g ${az_vm_resources_group} --vm-name ${az_vm_name} | jq -r .[0].id | cut -d "/" -f 9)
```

Comment:
Looks like we should use `az_resources_group` here? if it's set from L7 
(Similar for L38, 42 and 46)

Reply:
Good spot, I've updated it